code/modules/research/xenoarchaeology/finds/finds_eguns.dm

```diff
@@ -44,11 +39,39 @@
 /obj/item/gun/energy/captain/xenoarch/update_icon()
 	return
 
+//physical projectiles
 /obj/item/gun/projectile/revolver/xenoarch
 	name = "Fossilised Gun"
+	icon = 'icons/obj/xenoarchaeology.dmi'
+	max_shells = 12
 
 /obj/item/gun/projectile/revolver/xenoarch/refresh_upgrades()
 	return
 
 /obj/item/gun/projectile/revolver/xenoarch/update_icon()
 	return
+
+/obj/item/gun/projectile/revolver/sixshot/xenoarch
```

Comment:
being a revolver means that it can use *spin and make its icon invinsable

Reply:
oh did I forget to add the update_icon line to it? My bad. I tried to rework them to get the icons to appear like 6 times before I realized that the icons wont appear when spawned in normally due to them getting assigned sprites in the finds code. (which is dumb and I hate) Let me add the procs for them.

Reply:
its got the line for update sprite. It would be a bug that was present before I started in that case sense I used the exact same lines the normal revolver for xenoarch had before as base.